### PROBLEM STATEMENT

Detecting the type of Skin Disorder. Based on the given features, predict that which type of skin disorder a patient is having.

In [2]:
import matplotlib.pyplot as plt
import seaborn as sns
import pandas as pd
import numpy as np
%matplotlib inline
import warnings
warnings.filterwarnings('ignore')
from scipy import stats

In [3]:
data=pd.read_csv(r"C:\Users\Athul\Documents\DataScience_project\PRCP-1028-Skin-Disorder-Prediction\dataset_35_dermatology (1).csv")

In [4]:
data.head()

erythema  scaling  definite_borders  itching  koebner_phenomenon  \
0         2        2                 0        3                   0   
1         3        3                 3        2                   1   
2         2        1                 2        3                   1   
3         2        2                 2        0                   0   
4         2        3                 2        2                   2   

   polygonal_papules  follicular_papules  oral_mucosal_involvement  \
0                  0                   0                         0   
1                  0                   0                         0   
2                  3                   0                         3   
3                  0                   0                         0   
4                  2                   0                         2   

   knee_and_elbow_involvement  scalp_involvement  ...  \
0                           1                  0  ...   
1                           1                  1  ...   
2                           0                  0  ...   
3                           3                  2  ...   
4                           0                  0  ...   

   disappearance_of_the_granular_layer  \
0                                    0   
1                                    0   
2                                    0   
3                                    3   
4                                    2   

   vacuolisation_and_damage_of_basal_layer  spongiosis  \
0                                        0           3   
1                                        0           0   
2                                        2           3   
3                                        0           0   
4                                        3           2   

   saw-tooth_appearance_of_retes  follicular_horn_plug  \
0                              0                     0   
1                              0                     0   
2                              2                     0   
3                              0                     0   
4                              3                     0   

   perifollicular_parakeratosis  inflammatory_monoluclear_inflitrate  \
0                             0                                    1   
1                             0                                    1   
2                             0                                    2   
3                             0                                    3   
4                             0                                    2   

   band-like_infiltrate  Age  class  
0                     0   55      2  
1                     0    8      1  
2                     3   26      3  
3                     0   40      1  
4                     3   45      3  

[5 rows x 35 columns]

In [5]:
data.erythema.unique()

array([2, 3, 1, 0], dtype=int64)

## DOMAIN EXPERTISE

### Clinical Attributes

##### 1. Erythema
- Erythema or redness of the skin caused by injury or another inflammation-causing condition. Often presenting as a rash, erythema can be caused by environmental factors, infection, or overexposure to the sun (ie, sunburn).
- Values range from 0 to 3.
##### 2. Scaling
- Scales are a visible peeling or flaking of outer skin layers.
- Values range from 0 to 3.
##### 3. Definite Borders
- Uneven (irregular) borders, values range from 0 to 3.
##### 4. Itching
- Itching is an irritating sensation that makes you want to scratch your skin.
- Values range from 0 to 3.
##### 5. Koebner Phenomenon
- Koebner phenomenon describes the appearance of new skin lesions of a pre-existing dermatosis on areas of cutaneous injury in otherwise healthy skin.
- Values range from 0 to 3.
##### 6. Polygonal Papules
- Papules and polygonal plaques are shiny, flat-topped and firm on palpation.
- Values range from 0 to 3.
##### 7. Follicular Papules
- A follicular papule occurs when the inflammatory infiltrate accumulates within the epidermis of the follicle or in the perifollicular dermis.
- Values range from 0 to 3.
##### 8. Oral Mucosal Involvement
- The oral mucosa is the mucous membrane lining or “skin” inside of the mouth, including cheeks and lips.
- Values range from 0 to 3.
##### 9. Knee and Elbow Involvement
- Causes a rash with itchy, scaly patches, most commonly on the knees, elbows, trunk and scalp.
- Values range from 0 to 3.
##### 10. Scalp Involvement
- Scales patches, itchy on scalps; Value ranges from 0 to 3.
##### 11. Family History
- Whether diseases are observed in the family.
- Values 1 and 0, 1 if any of these diseases has been observed in the family and 0 if not.

### Histopathological Attributes

##### 12. Melanin Incontinence
- Melanin incontinence (pigmentary incontinence) is the presence of melanin in the superficial dermis, due to the loss of melanin from damaged cells of the basal layer.
- Values ranges from 0 to 3.
##### 13. Eosinophils in the Infiltrate
- Affects specific tissue layers or adnexal structures of the skin.
- Values ranges from 0 to 3.
##### 14. PNL Infiltrate
- Pure Neural Leprosy (PNL) is a rare clinical form of leprosy in which patients do not present with the classical skin lesions but have a high burden of the disability associated with the disease.
- Values ranges from 0 to 3.
##### 15. Fibrosis of the Papillary Dermis
- Fibroplasia of papillary dermis is a frequent histological finding present in dysplastic melanocytic nevi and melanomas.
- Values ranges from 0 to 3.
##### 16. Exocytosis
- Exocytosis is the process by which cells move materials from within the cell into the extracellular fluid.
- Values ranges from 0 to 3.
##### 17. Acanthosis
- Acanthosis nigricans is a velvety, darkening of the skin that usually occurs in intertriginous areas.
- Values ranges from 0 to 3.
##### 18. Hyperkeratosis
- Hyperkeratosis is a condition that causes your skin to thicken in certain places.
- Values ranges from 0 to 3.
##### 19. Parakeratosis
- Parakeratosis refers to incomplete maturation of epidermal keratinocytes, resulting in abnormal retention of nuclei in the stratum corneum.
- Values ranges from 0 to 3.
##### 20. Clubbing of the Rete Ridges
- Clubbed and evenly elongated rete ridges were more common in the Psoriasis group than in the Seboreic dermatitis groups.
- Values ranges from 0 to 3.
##### 21. Elongation of the Rete Ridges
- Psoriasis is the prototype of a group of cutaneous disorders (psoriasiform dermatitides) that show psoriasiform epidermal hyperplasia, defined as regular elongation of the rete ridges with preservation of the rete ridge-dermal papillae pattern.
- Values ranges from 0 to 3.
##### 22. Thinning of the Suprapapillary Epidermis
- This was defined as a thinning of the granular layer at the tips of the papillae.
- Values ranges from 0 to 3.
##### 23. Spongiform Pustule
- Spongiform pustule of Kogoj, due to neutrophils migrations from superficial dermal capillaries to intercellular spaces of the edematous Malpighian layer, leading to keratinocytes degeneration and formation of unilocular pustules.
- Values ranges from 0 to 3.
##### 24. Munro Microabcess
- Munro's microabscess is an abscess (collection of neutrophils) in the stratum corneum of the epidermis due to the infiltration of neutrophils from papillary dermis into the epidermal stratum corneum.
- Values ranges from 0 to 3.
##### 25. Focal Hypergranulosis
- Hypergranulosis is an increased thickness of the stratum granulosum.
- Values ranges from 0 to 3.
##### 26. Disappearance of the Granular Layer
- Parakeratosis is associated with the thinning or loss of the granular layer and is usually seen in diseases of increased cell turnover, whether inflammatory or neoplastic.
- Values ranges from 0 to 3.
##### 27.Vacuolisation and Damage of Basal Layer
- In dermatopathology "vacuolization" often refers specifically to vacuoles in the basal cell-basement membrane zone area, where it is an unspecific sign of disease.
- Values ranges from 0 to 3.
##### 28. Spongiosis
- Spongiosis is mainly intercellular edema (abnormal accumulation of fluid) in the epidermis, and is characteristic of eczematous dermatitis, manifested clinically by intraepidermal vesicles (fluid-containing spaces), "juicy" papules and/or lichenification.
- Values ranges from 0 to 3.
##### 29. Saw-tooth Appearance of Retes
- A 'sawtooth' pattern of the rete can be observed in oral lichen planus but this histologic finding is more common in cutaneous LP.
- Values ranges from 0 to 3.
##### 30. Follicular Horn Plug
- The precise cause or causes of follicular occlusion are unknown. There are genetic, hormonal and environmental influences.
- Values ranges from 0 to 3.
##### 31. Perifollicular Parakeratosis
- Unique epidermal reaction representing an acquired disorder of keratinization.
- Values ranges from 0 to 3.
##### 32. Inflammatory Monoluclear Inflitrate
- An inflammatory component is supported by the mononuclear cell infiltration in the myocardium as well as by the deposition of IgG, complement, and fibrin.
- Values ranges from 0 to 3.
##### 33. Band-like Infiltrate
- A bandlike infiltrate of inflammatory cells in the superficial dermis, parallel to the epidermis.
- Values ranges from 0 to 3.
##### 34. Age
- Represents the age of the patient
##### 35. Class (Target Variable)
- 1: Psoriasis - Psoriasis is a chronic disease characterized by inflammation with periods of exacerbation and remission, with the formation of erythematous plaques or papules with pearlescent-white scales on the skin. 
- 2: Seboreic Dermatitis - Seborrheic Dermatitis is a chronic inflammatory disease characterized by oily yellowish scaly and erythematous scaling patches, which can occur on the face, scalp and body folds, periodically recurs or heals. 
- 3: Lichen Planus - Lichen Planus is a papulosquamous, non-infectious inflammatory disease involving the skin and mucous membranes. It does not show organ involvement and involves skin, nails and mucosal surfaces. It is more common in women.
- 4: Pityriasis Rosea - Pityriasis rosea is a self-limiting, common acute papulosquamous inflammatory disease affecting the trunk and extremities
- 5: Chronic Dermatitis - It is a disease known as eczema in Turkey. Dermatitis types include atopic dermatitis, seborrheic dermatitis, contact dermatitis, allergic contact dermatitis, irritant contact dermatitis, hypereosinophilic dermatitis and other types. 
- 6: Pityriasis Rubra Pilaris - Pityriasis Rubra Pilaris (PRP) is a rare inflammatory disease, the etiology of which is not fully known, and the definitive treatment is still being investigated. 

In [7]:
data.describe()

erythema     scaling  definite_borders     itching  \
count  366.000000  366.000000        366.000000  366.000000   
mean     2.068306    1.795082          1.549180    1.366120   
std      0.664753    0.701527          0.907525    1.138299   
min      0.000000    0.000000          0.000000    0.000000   
25%      2.000000    1.000000          1.000000    0.000000   
50%      2.000000    2.000000          2.000000    1.000000   
75%      2.000000    2.000000          2.000000    2.000000   
max      3.000000    3.000000          3.000000    3.000000   

       koebner_phenomenon  polygonal_papules  follicular_papules  \
count          366.000000         366.000000          366.000000   
mean             0.633880           0.448087            0.166667   
std              0.908016           0.957327            0.570588   
min              0.000000           0.000000            0.000000   
25%              0.000000           0.000000            0.000000   
50%              0.000000           0.000000            0.000000   
75%              1.000000           0.000000            0.000000   
max              3.000000           3.000000            3.000000   

       oral_mucosal_involvement  knee_and_elbow_involvement  \
count                366.000000                  366.000000   
mean                   0.377049                    0.614754   
std                    0.834147                    0.982979   
min                    0.000000                    0.000000   
25%                    0.000000                    0.000000   
50%                    0.000000                    0.000000   
75%                    0.000000                    1.000000   
max                    3.000000                    3.000000   

       scalp_involvement  ...  focal_hypergranulosis  \
count         366.000000  ...             366.000000   
mean            0.519126  ...               0.393443   
std             0.905639  ...               0.849406   
min             0.000000  ...               0.000000   
25%             0.000000  ...               0.000000   
50%             0.000000  ...               0.000000   
75%             1.000000  ...               0.000000   
max             3.000000  ...               3.000000   

       disappearance_of_the_granular_layer  \
count                           366.000000   
mean                              0.464481   
std                               0.864899   
min                               0.000000   
25%                               0.000000   
50%                               0.000000   
75%                               1.000000   
max                               3.000000   

       vacuolisation_and_damage_of_basal_layer  spongiosis  \
count                               366.000000  366.000000   
mean                                  0.456284    0.953552   
std                                   0.954873    1.130172   
min                                   0.000000    0.000000   
25%                                   0.000000    0.000000   
50%                                   0.000000    0.000000   
75%                                   0.000000    2.000000   
max                                   3.000000    3.000000   

       saw-tooth_appearance_of_retes  follicular_horn_plug  \
count                     366.000000            366.000000   
mean                        0.453552              0.103825   
std                         0.954744              0.450433   
min                         0.000000              0.000000   
25%                         0.000000              0.000000   
50%                         0.000000              0.000000   
75%                         0.000000              0.000000   
max                         3.000000              3.000000   

       perifollicular_parakeratosis  inflammatory_monoluclear_inflitrate  \
count                    366.000000                           366.000000   
mean                       0.114754                             1.8661

In [8]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 366 entries, 0 to 365
Data columns (total 35 columns):
 #   Column                                    Non-Null Count  Dtype 
---  ------                                    --------------  ----- 
 0   erythema                                  366 non-null    int64 
 1   scaling                                   366 non-null    int64 
 2   definite_borders                          366 non-null    int64 
 3   itching                                   366 non-null    int64 
 4   koebner_phenomenon                        366 non-null    int64 
 5   polygonal_papules                         366 non-null    int64 
 6   follicular_papules                        366 non-null    int64 
 7   oral_mucosal_involvement                  366 non-null    int64 
 8   knee_and_elbow_involvement                366 non-null    int64 
 9   scalp_involvement                         366 non-null    int64 
 10  family_history                            366 non-

In [9]:
data.Age.unique()

array(['55', '8', '26', '40', '45', '41', '18', '57', '22', '30', '20',
       '21', '10', '65', '38', '23', '17', '51', '42', '44', '33', '43',
       '50', '34', '?', '15', '46', '62', '35', '48', '12', '52', '60',
       '32', '19', '29', '25', '36', '13', '27', '31', '28', '64', '39',
       '47', '16', '0', '7', '70', '37', '61', '67', '56', '53', '24',
       '58', '49', '63', '68', '9', '75'], dtype=object)

In [10]:
# Removing entries with Age = '?'
data = data.drop(data[data['Age'] == '?'].index)

In [11]:
data['Age']= data['Age'].astype(str).astype(int)

In [12]:
data.info()

<class 'pandas.core.frame.DataFrame'>
Index: 358 entries, 0 to 365
Data columns (total 35 columns):
 #   Column                                    Non-Null Count  Dtype
---  ------                                    --------------  -----
 0   erythema                                  358 non-null    int64
 1   scaling                                   358 non-null    int64
 2   definite_borders                          358 non-null    int64
 3   itching                                   358 non-null    int64
 4   koebner_phenomenon                        358 non-null    int64
 5   polygonal_papules                         358 non-null    int64
 6   follicular_papules                        358 non-null    int64
 7   oral_mucosal_involvement                  358 non-null    int64
 8   knee_and_elbow_involvement                358 non-null    int64
 9   scalp_involvement                         358 non-null    int64
 10  family_history                            358 non-null    int64
 11

In [13]:
data.rename(columns={'class':'Type'},inplace=True)

In [14]:
data.Type.unique()

array([2, 1, 3, 5, 4, 6], dtype=int64)

In [15]:
data["Type"].value_counts()

Type
1    111
3     71
2     60
5     48
4     48
6     20
Name: count, dtype: int64

In [16]:
data.Type=data.Type.map({1:0,2:1,3:2,4:3,5:4,6:5})
data.head()

erythema  scaling  definite_borders  itching  koebner_phenomenon  \
0         2        2                 0        3                   0   
1         3        3                 3        2                   1   
2         2        1                 2        3                   1   
3         2        2                 2        0                   0   
4         2        3                 2        2                   2   

   polygonal_papules  follicular_papules  oral_mucosal_involvement  \
0                  0                   0                         0   
1                  0                   0                         0   
2                  3                   0                         3   
3                  0                   0                         0   
4                  2                   0                         2   

   knee_and_elbow_involvement  scalp_involvement  ...  \
0                           1                  0  ...   
1                           1                  1  ...   
2                           0                  0  ...   
3                           3                  2  ...   
4                           0                  0  ...   

   disappearance_of_the_granular_layer  \
0                                    0   
1                                    0   
2                                    0   
3                                    3   
4                                    2   

   vacuolisation_and_damage_of_basal_layer  spongiosis  \
0                                        0           3   
1                                        0           0   
2                                        2           3   
3                                        0           0   
4                                        3           2   

   saw-tooth_appearance_of_retes  follicular_horn_plug  \
0                              0                     0   
1                              0                     0   
2                              2                     0   
3                              0                     0   
4                              3                     0   

   perifollicular_parakeratosis  inflammatory_monoluclear_inflitrate  \
0                             0                                    1   
1                             0                                    1   
2                             0                                    2   
3                             0                                    3   
4                             0                                    2   

   band-like_infiltrate  Age  Type  
0                     0   55     1  
1                     0    8     0  
2                     3   26     2  
3                     0   40     0  
4                     3   45     2  

[5 rows x 35 columns]

## EXPLORATORY DATA ANALYSIS

### Univariate Analysis

In [19]:
#!pip install sweetviz

In [20]:
import sweetviz  as sv
my_report=sv.analyze(data)
my_report.show_html()

                                             |          | [  0%]   00:00 -> (? left)

Report SWEETVIZ_REPORT.html was generated! NOTEBOOK/COLAB USERS: the web browser MAY not pop up, regardless, the report IS saved in your notebook/colab files.


### Bivariate Analysis

In [22]:
data_1=data[["erythema", "scaling", "definite_borders", "itching", "koebner_phenomenon", 
                        "polygonal_papules", "follicular_papules", "oral_mucosal_involvement", 
                        "knee_and_elbow_involvement","scalp_involvement", "family_history"]]
data_1

erythema  scaling  definite_borders  itching  koebner_phenomenon  \
0           2        2                 0        3                   0   
1           3        3                 3        2                   1   
2           2        1                 2        3                   1   
3           2        2                 2        0                   0   
4           2        3                 2        2                   2   
..        ...      ...               ...      ...                 ...   
361         2        1                 1        0                   1   
362         3        2                 1        0                   1   
363         3        2                 2        2                   3   
364         2        1                 3        1                   2   
365         3        2                 2        0                   0   

     polygonal_papules  follicular_papules  oral_mucosal_involvement  \
0                    0                   0                         0   
1                    0                   0                         0   
2                    3                   0                         3   
3                    0                   0                         0   
4                    2                   0                         2   
..                 ...                 ...                       ...   
361                  0                   0                         0   
362                  0                   0                         0   
363                  2                   0                         2   
364                  3                   0                         2   
365                  0                   0                         0   

     knee_and_elbow_involvement  scalp_involvement  family_history  
0                             1                  0               0  
1                             1                  1               1  
2                             0                  0               0  
3                             3                  2               0  
4                             0                  0               0  
..                          ...                ...             ...  
361                           0                  0               0  
362                           0                  0               0  
363                           0                  0               0  
364                           0                  0               0  
365                           3                  3               0  

[358 rows x 11 columns]

In [ ]:
sns.pairplot(data)

In [ ]:
plt.figure(figsize=(35, 35))#canvas size
sns.heatmap(data_1.corr(), annot=True, cmap="RdYlGn", annot_kws={"size":15})

In [ ]:
plt.figure(figsize=(30,30),facecolor='white')
plotnumber=1
for column in data_1.columns:
    if plotnumber<=12:
        ax=plt.subplot(4,3,plotnumber)
        sns.histplot(x=data_1[column],hue=data.Type)
        plt.xlabel(column,fontsize=30)
        plt.ylabel('Type',fontsize=30)
    plotnumber+=1
plt.tight_layout()

#### INSIGHTS :

- Melanin_incontinence is mostly seen in Lichen Planus.
- Eosinophils_in_the_infiltrate are least in Psoriasis and slightly seen in Seboreic Dermatitis.
- PNL_infiltrate is least seen in Seboreic Dermatitis and Cronic Dermatitis.
- Fibrosis_of_the_papillary_dermis are least in Psoriasis,Seboreic Dermatitis,Lichen Planus,Pityriasis Rosea and Pityriasis Rubra Pilaris and mostly seen in Cronic Dermatitis.
- Exocytosis least in Psoriasis and majorly in Lichen Planus.
- Acanthosis are fairly seen in Psoriasis,Lichen Planus and cronic Deramatitis.
- Only few Psoriasis(4) and Seboreic Dermatitis(1) patiensts have Hyperkeratosis in high range.
- Paraketosis are fairly seen in Psoriasis,Lichen Planus.
- Clubbing_of_the_rete_ridges are least in Seboreic Dermatitis,Lichen Planus,Pityriasis Rosea and majorly in Psoriasis.
- Elongation_of_the_rete_ridges are majorly seen in Psoriasis,Cronic Dermititis.
- Thinning_of_the_suprapapillary_epidermis  are mostly seen in Psoriasis.
- spongiform_pustule are slightly present in Psoriasis and least in other classes.
- Few Psoriasis patients have  Munro_microabacess and rest classes have lower range.
- Focal_hypergranulosis are slightly seen in Lichen Planus.
- Disappearance_of_the_granular_layer minimally seen in Psoriasis.
- Vacuolisation_and_damage_of_basal_layer mostly seen in Lichen Planus.
- Spongiosis are least in Psoriais.
- Saw-tooth_appearance_of_retes is primarly seen in Lichen Planus.
- Follicular_horn_plug are seen Pityriasis Rubra Pilaris and least in other classes.
- Perifollicular_parakeratosis also slightly found in Pityriasis Rubra Pilaris.
- Inflammatory_monoluclear_inflitrate actually seen in every classes.
- Band-like_infiltrate majorly seen in Lichen Planus.

## MODEL BUILDING


### Logistic Regression Model

In [ ]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import confusion_matrix,accuracy_score,recall_score,precision_score,classification_report,f1_score,roc_auc_score

# Define the features and the target variable
X = data.iloc[:, :-1].values
y = data.iloc[:, -1].values

In [ ]:
from sklearn.preprocessing import StandardScaler
sc=StandardScaler()
X_new=sc.fit_transform(X)

In [ ]:
from sklearn.model_selection import train_test_split

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.25, random_state=4)

In [ ]:
# Create a logistic regression classifier
lr_clf=LogisticRegression()
lr_clf.fit(X_train,y_train)
y_pred_LR=lr_clf.predict(X_test)

In [ ]:
cm=confusion_matrix(y_test,y_pred_LR)
print(cm)

In [ ]:
pd.crosstab(y_test,y_pred_LR)

In [ ]:
print(classification_report(y_test,y_pred_LR))

### Random Forest

In [ ]:
from sklearn.ensemble import RandomForestClassifier#importing randomforest

rf_clf = RandomForestClassifier(n_estimators=100)#object creation ,taking 100 decision tree in random forest 
rf_clf.fit(X_train,y_train)#training the data

In [ ]:
y_pred_RF=rf_clf.predict(X_test)

In [ ]:
print(classification_report(y_test,y_pred_RF))

### XGBoost

In [ ]:
#!pip install xgboost

In [ ]:
import xgboost as xgb
# Create an XGBoost classifier
xgb_classifier = xgb.XGBClassifier(use_label_encoder=False, eval_metric='mlogloss')

# Train the XGBoost classifier
xgb_classifier.fit(X_train, y_train)

# Make predictions on the test set
Y_pred_xgb = xgb_classifier.predict(X_test)

In [ ]:
print(classification_report(y_test,Y_pred_xgb ))

### Decision Tree

In [ ]:
from sklearn.tree import DecisionTreeClassifier
dt = DecisionTreeClassifier()
dt.fit(X_train,y_train)

In [ ]:
y_pred_dt = dt.predict(X_test)
print(classification_report(y_test,y_pred_dt))

### Support Vector Machine

In [ ]:
from sklearn.svm import SVC
svc = SVC()
svc.fit(X_train,y_train)
y_pred_svc = svc.predict(X_test)

In [ ]:
print(classification_report(y_test,y_pred_svc))

## PREDICTING THE TYPE OF DISEASE

In [ ]:
print("Accuracy of Logistic Regression: ", accuracy_score(y_test,y_pred_LR))
print("Accuracy of Random Forest: ", accuracy_score(y_test,y_pred_RF))
print("Accuracy of XGBoost: ", accuracy_score(y_test,Y_pred_xgb))
print("Accuracy of Decision Tree: ", accuracy_score(y_test,y_pred_dt))
print("Accuracy of Support Vector Machine: ", accuracy_score(y_test,y_pred_svc))

#### We can see that the highest accuracy is seen in Random Forest model and Logistic Regression

In [ ]:
Erythema= int(input('value of Erythema:'))
Scaling= int(input('value of Scaling:'))
Definite_Borders= int(input('value of Definite Borders:'))
Itching= int(input('value of Itching:'))
Koebner_Phenomenon= int(input('value of Koebner Phenomenon:'))
Polygonal_Papules= int(input('value of Polygonal Papules:'))
Follicular_Papules= int(input('value of Follicular Papules:'))
Oral_Mucosal_Involvement= int(input('value of Oral Mucosal Involvement:'))
Knee_and_Elbow_Involvement= int(input('value of Knee and Elbow Involvement:'))
Scalp_Involvement= int(input('value of Scalp Involvement:'))
Family_History= int(input('value of Family History:'))
Melanin_Incontinence= int(input('value of Melanin Incontinence:'))
Eosinophils_in_the_Infiltrate= int(input('value of Eosinophils in the Infiltrate:'))
PNL_Infiltrate= int(input('value of PNL Infiltrate:'))
Fibrosis_of_the_Papillary_Dermis= int(input('value of Fibrosis of the Papillary Dermis:'))
Exocytosis= int(input('value of Exocytosis:'))
Acanthosis= int(input('value of Acanthosis:'))
Hyperkeratosis= int(input('value of Hyperkeratosis:'))
Parakeratosis= int(input('value of Parakeratosis:'))
Clubbing_of_the_Rete_Ridges= int(input('value of Clubbing of the Rete Ridges:'))
Elongation_of_the_Rete_Ridges= int(input('value of Elongation of the Rete Ridges:'))
Thinning_of_the_Suprapapillary_Epidermis= int(input('value of Thinning of the Suprapapillary Epidermis:'))
Spongiform_Pustule= int(input('value of Spongiform Pustule:'))
Munro_Microabcess= int(input('value of Munro Microabcess:'))
Focal_Hypergranulosis= int(input('value of Focal Hypergranulosis:'))
Disappearance_of_the_Granular_Layer= int(input('value of Disappearance of the Granular Layer:'))
Vacuolisation_and_Damage_of_Basal_Layer= int(input('value of Vacuolisation and Damage of Basal Layer:'))
Spongiosis= int(input('value of Spongiosis:'))
Saw_tooth_Appearance_of_Retes= int(input('value of Saw-tooth Appearance of Retes:'))
Follicular_Horn_Plug= int(input('value of Follicular Horn Plug:'))
Perifollicular_Parakeratosis= int(input('value of Perifollicular Parakeratosis:'))
Inflammatory_Monoluclear_Inflitrate= int(input('value of Inflammatory Monoluclear Inflitrate:'))
Band_like_Infiltrate= int(input('value of Band-like Infiltrate:'))
Age= int(input('Age:'))

S =[Erythema,Scaling,Definite_Borders,Itching,Koebner_Phenomenon,Polygonal_Papules,Follicular_Papules,Oral_Mucosal_Involvement,Knee_and_Elbow_Involvement,Scalp_Involvement,Family_History,Melanin_Incontinence,Eosinophils_in_the_Infiltrate,PNL_Infiltrate,Fibrosis_of_the_Papillary_Dermis,Exocytosis,Acanthosis,Hyperkeratosis,Parakeratosis,Clubbing_of_the_Rete_Ridges,Elongation_of_the_Rete_Ridges,Thinning_of_the_Suprapapillary_Epidermis,Spongiform_Pustule,Munro_Microabcess,Focal_Hypergranulosis,Disappearance_of_the_Granular_Layer,Vacuolisation_and_Damage_of_Basal_Layer,Spongiosis,Saw_tooth_Appearance_of_Retes,Follicular_Horn_Plug,Perifollicular_Parakeratosis,Inflammatory_Monoluclear_Inflitrate,Band_like_Infiltrate,Age]

p =rf_clf.predict([S])
print(p)
if (p==1):
    print("The person is having Psoriasis")
elif (p==2):
    print("The person is having Seboreic Dermatitis")
elif (p==3):
    print("The person is having Lichen Planus")
elif (p==4):
    print("The person is having Pityriasis Rosea")
elif (p==5):
    print("The person is having Chronic Dermatitis")
else:
    print("The person is having Pityriasis Rubra Pilaris")

## CONCLUSION :-
 In the field of dermatology, differential diagnosis of Erythmato-Squamous Diseases (ESD) are difficult. The six important types of ESDs are including Psoriasis, Seboreic dermatitis, Lichen planus, Pityriasis rosea, Chronic dermatitis and Pityriasis rubra pilaris.
 
 * The dataset contains 366 rows and 35 columns in which 11 features are clinically evaluated and 23 are found by taking the skin samples.
 * There was no null values but Age attribute had some wrong value "?" and also age attribute had 0 as a value, which is not possible  thus we changed the wrong value and replaced the 0 value with the mean.
* While checking the correlation of these attributes it is seen that many attributes have high accuracy which was leading for overfitting and thus we changed those as well. 
The dropped attributes are polygonal_papules, oral_mucosal_involvement, melanin_incontinence, focal_hypergranulosis, vacuolisation_and_damage_of_basal_layer, saw-tooth_appearance_of_retes and follicular_horn_plug.
* We are used 4 Machine Learning algorithms :- Logistic Regression,Decision Tree, Support Vector Machine, XGBoost and Random Forest.
* From these algorithms, Random Forest model is used for the prediction of the new data.

## SUGGESTION :-
Early detection is the best prevention. Both clinical and histopathological analysis is required for the detection. Eventhough our model gives accurate prediction of the disease it is also required to examine under a specialist doctor. In this model we had only used 28 attributes for prediction which can make the detection faster with the help of a specialist doctor.